src/traverse/mod.rs

```diff
@@ -450,69 +451,135 @@ pub(crate) fn extract(
 
     log::debug!("Running on absolute dirs {flow:?}");
 
-    // stage 2 - check for manifest, .rs , .md files and directories
     let mut files_to_check = Vec::with_capacity(64);
-    while let Some(path) = flow.pop_front() {
-        let x = if let Ok(meta) = path.metadata() {
-            if meta.is_file() {
-                match path.file_name().and_then(|x| x.to_str()) {
-                    Some(file_name) if file_name == "Cargo.toml" => Extraction::Manifest(path),
-                    Some(file_name) if file_name.ends_with(".md") => Extraction::Markdown(path),
-                    Some(file_name) if file_name.ends_with(".rs") => Extraction::Source(path),
-                    _ => {
-                        // This branch is commonly entered when ran on a non-cargo
-                        // path.
-                        // Potentially become mdbook aware
-                        // <https://github.com/drahnr/cargo-spellcheck/issues/273>
-                        log::debug!(
-                            "Unknown file type encountered, skipping path: {}",
-                            path.display()
-                        );
-                        continue;
+
+    // stage 2 - check for manifest, .rs , .md files and directories
+    if gitignore {
+        let max_depth = match recurse {
+            true => None,
+            false => Some(1),
+        };
+
+        let mut all_files = VecDeque::<PathBuf>::with_capacity(32);
+
+        while let Some(path) = flow.pop_front() {
+            let files = ignore::WalkBuilder::new(path)
```

Comment:
Can we define a closure that gives us an iterator back rather than duplicating the `if` and `else` branch logic. It's been a mess before you came, but now I can see what I've done a few years back :cold_sweat: